### DISCUSSION TITLE
Setting up blob storage for the planetary computer

### DISCUSSION THREAD

**Discussion by tristangrupp** [microsoft/PlanetaryComputer#190] [Q&A]
Hello! I have some rasters and geometries that I would like to use with datasets in the Planetary Computer catalog. How do I set up blob storage for these rasters and geometries to use in the Planetary Computer platform (say, in the R environment)? 

I am not sure how to initially get this set up.
Thanks!

**Comment by TomAugspurger** [comment]
At a high-level, you'll need a few things:

1. An Azure Subscription (you can use a [free trial](https://azure.microsoft.com/en-in/offers/ms-azr-0044p/) to test stuff out)
2. An [Azure Blob Storage](https://learn.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction) storage account and storage container for hosting your rasters and geometries.
3. Some way to get your rasters and geometries to your storage container
   - [azcopy](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10) might be a good option for one-time uploads

Once you have all those prerequisites in place, the only additional hurdle to using those assets is getting credentials for your storage containers into the compute environment (it sounds like you're using the Hub). We have an example at https://planetarycomputer.microsoft.com/docs/quickstarts/storage/ (in Python) that uses SAS tokens that relies on pasting in a SAS token to a kernel using `getpass` Just note that we recommend *not* storing sensitive information like a Azure credentials in your Hub home directory.

If you setup compute in your own Azure Subscription (instead of using the Hub), then you'd have access to all the ways to authenticate with Blob Storage (e.g. https://learn.microsoft.com/en-us/azure/developer/python/sdk/authentication-overview)
